src/script/entity/message/File.ts

```diff
@@ -91,35 +104,35 @@ export class File extends Asset {
 
   /**
    * Loads and decrypts otr asset as initiates download
-   *
-   * @returns {Promise} Returns a promise that resolves with the asset as blob
    */
-  download() {
+  download(): Promise<number | void> {
     if (this.status() !== AssetTransferState.UPLOADED) {
       return Promise.resolve(undefined);
     }
 
     const download_started = Date.now();
 
     return this.load()
-      .then(blob => downloadBlob(blob, this.file_name))
-      .then(() => {
+      .then(blob => {
+        if (!blob) {
+          throw new Error('No blob received.');
+        }
+        return downloadBlob(blob, this.file_name);
+      })
+      .then(blob => {
         const download_duration = (Date.now() - download_started) / TimeUtil.UNITS_IN_MILLIS.SECOND;
         this.logger.info(`Downloaded asset in ${download_duration} seconds`);
+        return blob;
       })
       .catch(error => this.logger.error('Failed to download asset', error));
   }
 
-  cancel_download() {
+  cancel_download(): void {
     this.status(AssetTransferState.UPLOADED);
     return this.original_resource().cancelDownload();
```

Comment:
Let's maybe remove the `return` here that doesn't serve any purposes (as shown by the signature)